resolver/resolver.go

```diff
@@ -457,7 +457,12 @@ func isUDP(w dns.ResponseWriter) bool {
 // truncate removes answers until the given dns.Msg fits the permitted
 // length of the given transmission channel and sets the TC bit.
 // See https://tools.ietf.org/html/rfc1035#section-4.2.1
-func truncate(m *dns.Msg, udp bool) *dns.Msg {
+// If `setTruncateBit` is true, the Truncate bit will be set if the message
+// has been truncated already or gets truncated here.
+// If `setTruncateBit` is false, the message will have its Truncate bit
+// cleared if it is already set, and won't set it even if it does get
+// truncated.
+func truncate(m *dns.Msg, udp bool, setTruncateBit bool) *dns.Msg {
```

Comment:
I believe c-style code deserve the full treatment, including c-style function signatures. The author might really like one-liners, but I don't think it's worth sacrificing clarity. It's your call though.